it/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnectorSpec.scala

```diff
@@ -0,0 +1,139 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementseisstub.connectors
+
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import com.github.tomakehurst.wiremock.client.WireMock._
+import com.github.tomakehurst.wiremock.matching.AnythingPattern
+import com.github.tomakehurst.wiremock.matching.StringValuePattern
+import org.mockito.ArgumentMatchers
+import org.mockito.MockitoSugar
+import org.scalacheck.Gen
+import org.scalatest.concurrent.IntegrationPatience
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import play.api.http.HeaderNames
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.test.HttpClientV2Support
+import uk.gov.hmrc.transitmovementseisstub.config.EISInstanceConfig
+import uk.gov.hmrc.transitmovementseisstub.config.Headers
+import uk.gov.hmrc.transitmovementseisstub.connectors.errors.RoutingError
+import uk.gov.hmrc.transitmovementseisstub.utils.FakeRequestBuilder
+import uk.gov.hmrc.transitmovementseisstub.utils.TestActorSystem
+import uk.gov.hmrc.transitmovementseisstub.utils.TestHelpers
+import uk.gov.hmrc.transitmovementseisstub.utils.WiremockSuite
+
+import java.net.URL
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class EISConnectorSpec
+    extends AnyWordSpec
+    with HttpClientV2Support
+    with Matchers
+    with WiremockSuite
+    with ScalaFutures
+    with MockitoSugar
+    with IntegrationPatience
+    with ScalaCheckPropertyChecks
+    with TableDrivenPropertyChecks
+    with TestActorSystem {
+
+  lazy val anything: StringValuePattern = new AnythingPattern()
+
+  val uriStub = "/ncts5/traderchannelsubmissionsgb/v1"
+
+  val connectorConfig: EISInstanceConfig = EISInstanceConfig(
+    "http",
+    "localhost",
+    wiremockPort,
+    uriStub,
+    Headers("bearertokenhereGB")
+  )
+
+  // We construct the connector each time to avoid issues with the circuit breaker
+  def connector = new EISConnectorImpl("eis", connectorConfig, httpClientV2)
+
+  def source: Source[ByteString, _] = Source.single(ByteString.fromString("<test></test>"))
+
+  def stub(codeToReturn: Int, body: String = "") =
+    server.stubFor(
+      post(
+        urlEqualTo(uriStub)
+      )
+        .willReturn(aResponse().withStatus(codeToReturn).withBody(body))
+    )
+
+  "post" should {
+
+    "returns a Right of unit if the call is successful" in {
+      server.resetAll()
+
+      stub(OK)
+
+      val hc = HeaderCarrier()
+
+      whenReady(connector.post(source)(hc)) {
+        _.isRight mustBe true
+      }
```

Comment:
We need to ensure that the stub passes the auth token that it received through to EIS (so, by extension, the config should not need the bearer token specified in it).

So, test the stub has the appropriate `Authorization` header, which should match that in the header carrier's `authorization` field.

Reply:
good point, removed the config and added test to ensure it's passed on from the HeaderCarrier